README.md
# Dialog

[![npm package](https://img.shields.io/npm/v/@_nu/vue-dialog.svg)](https://www.npmjs.org/package/@_nu/vue-dialog)
[![github](https://img.shields.io/github/stars/nu-system/vue-dialog.svg?style=social)](https://github.com/nu-system/vue-dialog)

<iframe src="https://codesandbox.io/embed/nudialogvue-phc9q?autoresize=1&fontsize=14&hidenavigation=1&view=preview" title="nu-dialog-vue" allow="geolocation; microphone; camera; midi; vr; accelerometer; gyroscope; payment; ambient-light-sensor; encrypted-media" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>

组件库母版系统 [NU-system](https://nu-system.github.io/) 弹窗组件 VUE 实现。

`nu-vue-dialog` 本身不会输出任何样式，基础样式来自于 [nu-dialog](https://nu-system.github.io/vanilla/dialog/) , 

## 怎么用？

```bash
$ yarn add @_nu/vue-dialog @_nu/vanilla-dialog
```

### 二次封装

```vue

<script>
  import NuDialog from "@_nu/vue-dialog"
  export default NuDialog;
</script>

<!--样式引用-->
<style src="@_nu/vanilla-dialog"></style>
<style src="@_nu/vanilla-dialog/css/position/middle.css"></style>
<!-- 
<style src="@_nu/vanilla-dialog/css/position/top.css"></style>
<style src="@_nu/vanilla-dialog/css/position/left.css"></style>
<style src="@_nu/vanilla-dialog/css/position/right.css"></style>
<style src="@_nu/vanilla-dialog/css/position/bottom.css"></style> 
-->
```


### 使用

```vue
<template>
  <div id="app">    
    <button type="button" @click="dialogOpen = true">打开弹窗</button>            
    <Dialog :open.sync="dialogOpen" :position="dialogPosition">
      <select @change="handlePosition">
        <option value="middle">居中显示</option>
        <option value="top">居上显示</option>
        <option value="left">居左显示</option>
        <option value="right">居右显示</option>
        <option value="bottom">居下显示</option>
      </select>
    </Dialog>
  </div>
</template>

<script>
  import Dialog from "@components/Dialog";
  
  export default {
    name: 'app',
    data() {
      return {
        dialogOpen: false,
        dialogPosition: 'middle'
      }
    },
    components: {
      Dialog
    },
    methods: {
        handlePosition(e) {
         this.dialogPosition = e.target.value;
        }
    }
}
</script>
```

nu-dialog-vue 会动态的把弹窗添加到 `body` 标签之后。

## Api

| props   |      类型      | 默认值  |功能 |
|:----------|:-------------|:------:|------:|
| :open.sync |  boolean | - | 显示弹窗|
| :position |  strong | 'middle' | 弹窗位置|
| :beforeClose |  Func | - | 在关闭之前要做的事 |
| :isPortal | boolean | 'true' | 是否需要传送门 |
| :speed | Number | 200 | 动画时长 |

**position 可选值**: `middle`,`top`,`right`,`left`,`bottom`;

**beforeClose**: 如果返回值为 `false` 那么弹窗不会关闭;

## Dom 结构

```vue
<div v-if="render" class="nu_dialog_wrap" :class="computedClass">
  <slot name="mask">
    <label class="nu_dialog_mask" @click="handleClickMask"/>
  </slot>
  <slot name="dialog">
    <div class="nu_dialog">
      <slot name="close">
        <label class="nu_dialog_close" @click="handleClickClose">
          <slot name="close-icon">&times;</slot>
        </label>
      </slot>
      <slot></slot>
    </div>
  </slot>
</div>
```

`nu-dialog-vue` 几乎所有都子组件都可以用 `slot` 重写。


CHANGELOG.md
## [0.0.8](https://github.com/nu-system/vue-dialog/compare/v0.0.5...v0.0.8) (2019-09-02)


### Bug Fixes

* recover the isPortal what I have removed before ([75dd012](https://github.com/nu-system/vue-dialog/commit/75dd012))



## [0.0.5](https://github.com/nu-system/vue-dialog/compare/v0.0.4...v0.0.5) (2019-07-12)



## 0.0.4 (2019-07-12)





__test__/index.test.js
import Vue from 'vue'
import { mount, shallowMount } from '@vue/test-utils'
import { render } from '@vue/server-test-utils'
import NuDialog from '../src'
import Demo from './demo'
import sinon from 'sinon'
import FakeTimers from '@jest/fake-timers/build/jestFakeTimers';
import { wrap } from 'module';

 
describe('NuDialog开关问题', () => {
    it('is a Vue instance', () => {
        const wrapper = mount(NuDialog)
        expect(wrapper.isVueInstance()).toBeTruthy()
    })
    //open
    it('初始化为关闭的状态', async () => {
        const wrapper = mount(NuDialog,{
            propsData:{
                open:false
            }
        })
        wrapper.setProps({open:true})
        //是否打开
        expect(document.body.className).toEqual(expect.stringContaining("nu_dialog_open"))
        expect(wrapper.vm.show).toEqual(false)
        expect(wrapper.vm.render).toEqual(true)
        await new Promise((rs)=>{
            setTimeout(()=>{
                expect(wrapper.vm.show).toEqual(true)
                expect(wrapper.vm.render).toEqual(true)
                rs()
            },16)
        })
        // // 关闭
        wrapper.setProps({open:false})
        expect(document.body.className).not.toEqual(expect.stringContaining("nu_dialog_open"))
        expect(wrapper.vm.show).toEqual(false)
        expect(wrapper.vm.render).toEqual(true)
        await new Promise((rs)=>{
            setTimeout(()=>{
                expect(wrapper.vm.show).toEqual(false)
                expect(wrapper.vm.render).toEqual(false)
                rs()
            },200)
        })
    })
    it('初始化为打开的状态',async ()=>{
        const wrapper = mount(NuDialog,{
            propsData:{
                open:true,
            }
        })
        // 关闭
        await new Promise((rs)=>{
            setTimeout(()=>{
                wrapper.setProps({open:false})
                expect(document.body.className).not.toEqual(expect.stringContaining("nu_dialog_open"))
                expect(wrapper.vm.show).toEqual(false)
                expect(wrapper.vm.render).toEqual(true)
                setTimeout(()=>{
                    expect(wrapper.vm.show).toEqual(false)
                    expect(wrapper.vm.render).toEqual(false)
                    rs()
                },200)
            },16)
        })
        // 打开
        wrapper.setProps({open:true})
        expect(document.body.className).toEqual(expect.stringContaining("nu_dialog_open"))
        expect(wrapper.vm.show).toEqual(false)
        expect(wrapper.vm.render).toEqual(true)
        await new Promise((rs)=>{
            setTimeout(()=>{
                expect(wrapper.vm.show).toEqual(true)
                expect(wrapper.vm.render).toEqual(true)
                rs()
            },16)
        })
        
    })
    // //position
    
    it('dialog position', async () => {
        const wrapper = mount(NuDialog)
        expect(wrapper.classes("_middle")).toEqual(true)
        wrapper.setProps({"position":"top"})
        expect(wrapper.classes("_top")).toEqual(true)
        wrapper.setProps({"position":"left"})
        expect(wrapper.classes("_left")).toEqual(true)
        wrapper.setProps({"position":"right"})
        expect(wrapper.classes("_right")).toEqual(true)
        wrapper.setProps({"position":"bottom"})
        expect(wrapper.classes("_bottom")).toEqual(true)
        expect(wrapper.classes("_middle")).toEqual(false)
    })

    //speed
    it('关闭动画速度',async ()=>{
        const wrapper = mount(NuDialog,{
            propsData:{
                open:true,
                speed:500
            }
        })
        //  关闭弹窗
        await new Promise((rs,rj)=>{
            setTimeout(()=>{
                wrapper.setProps({open:false})
                setTimeout(()=>{
                    expect(wrapper.vm.show).toEqual(false)
                    expect(wrapper.vm.render).toEqual(true)
                    setTimeout(()=>{
                        expect(wrapper.vm.show).toEqual(false)
                        expect(wrapper.vm.render).toEqual(false)
                        rs()
                    },400)
                },100)
            },16)
        })
    })
    it('用户快速快关',async ()=>{
        const wrapper = mount(NuDialog)
        wrapper.setProps({open:true})
        wrapper.setProps({open:false})
        wrapper.setProps({open:true})
        wrapper.setProps({open:true})
        wrapper.setProps({open:false})
        await new Promise((rs,rj)=>{
            setTimeout(()=>{
                expect(wrapper.vm.show).toEqual(false)
                expect(wrapper.vm.render).toEqual(false)
                rs()
            },300)
        })
    })
})

describe('beforeClose api',()=>{
    it('beforeClose触发成功', async () => {
        let invoke=false
        const wrapper = mount(NuDialog,{
            propsData:{
                open:true,
                speed:100,
                beforeClose:()=>{
                    invoke=true
                }
            }
        })
        await new Promise((rs,rj)=>{
            setTimeout(()=>{
                wrapper.setProps({open:false})
                expect(invoke).toBe(true)
                rs()
            },16)
        })
    })
    it('beforeClose返回false时，不关闭dialog', async () => {
        const wrapper = mount(NuDialog,{
            propsData:{
                open:true,
                speed:100,
                beforeClose:()=>{
                    return false
                }
            }
        })
        await new Promise((rs,rj)=>{
            setTimeout(()=>{
                // 这里一定要在open动作结束之后再操作
                wrapper.setProps({open:false})
                setTimeout(()=>{
                    expect(document.body.className).toEqual(expect.stringContaining("nu_dialog_open"))
                    rs()
                },100)
            },16)
        })
    })
})

describe('基于父元素 NuDialog trigger button', () => {
    it('close button的触发', async () => {
        const wrapper = mount(Demo)
        wrapper.find("button").trigger("click")
        expect(document.body.className).toBe("nu_dialog_open")
        wrapper.find(".nu_dialog_close").trigger("click")
        expect(document.body.className).toBe("")
    })
    it('mask的触发', async () => {
        const wrapper = mount(Demo)
        const dialog=wrapper.find(".nu_dialog_wrap")
        wrapper.find("button").trigger("click")
        expect(document.body.className).toBe("nu_dialog_open")
        wrapper.find(".nu_dialog_mask").trigger('click')
        expect(document.body.className).toBe("")
        
    })
})

describe('基于父元素 销毁dialog', () => {
    it('destory',  () => {
        const spy = sinon.stub()
        mount(NuDialog,{
            destroyed() {
                spy()
            }
        }).destroy()
        expect(spy.calledOnce).toBe(true)
    })
})

jest.config.js
// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html

module.exports = {
  // All imported modules in your tests should be mocked automatically
  // automock: false,

  // Stop running tests after `n` failures
  // bail: 0,

  // Respect "browser" field in package.json when resolving modules
  // browser: false,

  // The directory where Jest should store its cached dependency information
  // cacheDirectory: "/private/var/folders/f3/p_8cqzpx6q35q4wkwbzdnf1r0000gn/T/jest_dx",

  // Automatically clear mock calls and instances between every test
  clearMocks: true,

  // Indicates whether the coverage information should be collected while executing the test
  // collectCoverage: false,

  // An array of glob patterns indicating a set of files for which coverage information should be collected
  // collectCoverageFrom: null,

  // The directory where Jest should output its coverage files
  coverageDirectory: "coverage",

  // An array of regexp pattern strings used to skip coverage collection
  // coveragePathIgnorePatterns: [
  //   "/node_modules/"
  // ],

  // A list of reporter names that Jest uses when writing coverage reports
  // coverageReporters: [
  //   "json",
  //   "text",
  //   "lcov",
  //   "clover"
  // ],

  // An object that configures minimum threshold enforcement for coverage results
  // coverageThreshold: null,

  // A path to a custom dependency extractor
  // dependencyExtractor: null,

  // Make calling deprecated APIs throw helpful error messages
  // errorOnDeprecated: false,

  // Force coverage collection from ignored files using an array of glob patterns
  // forceCoverageMatch: [],

  // A path to a module which exports an async function that is triggered once before all test suites
  // globalSetup: null,

  // A path to a module which exports an async function that is triggered once after all test suites
  // globalTeardown: null,

  // A set of global variables that need to be available in all test environments
  // globals: {},

  // An array of directory names to be searched recursively up from the requiring module's location
  // moduleDirectories: [
  //   "node_modules"
  // ],

  // An array of file extensions your modules use
    moduleFileExtensions: [
      "js",
      "json",
      "vue"
    ],

  // A map from regular expressions to module names that allow to stub out resources with a single module
  // moduleNameMapper: {},

  // An array of regexp pattern strings, matched against all module paths before considered 'visible' to the module loader
  // modulePathIgnorePatterns: [],

  // Activates notifications for test results
  // notify: false,

  // An enum that specifies notification mode. Requires { notify: true }
  // notifyMode: "failure-change",

  // A preset that is used as a base for Jest's configuration
  // preset: null,

  // Run tests from one or more projects
  // projects: null,

  // Use this configuration option to add custom reporters to Jest
  // reporters: undefined,

  // Automatically reset mock state between every test
  // resetMocks: false,

  // Reset the module registry before running each individual test
  // resetModules: false,

  // A path to a custom resolver
  // resolver: null,

  // Automatically restore mock state between every test
  // restoreMocks: false,

  // The root directory that Jest should scan for tests and modules within
  // rootDir: null,

  // A list of paths to directories that Jest should use to search for files in
  // roots: [
  //   "<rootDir>"
  // ],

  // Allows you to use a custom runner instead of Jest's default test runner
  // runner: "jest-runner",

  // The paths to modules that run some code to configure or set up the testing environment before each test
  // setupFiles: [],

  // A list of paths to modules that run some code to configure or set up the testing framework before each test
  // setupFilesAfterEnv: [],

  // A list of paths to snapshot serializer modules Jest should use for snapshot testing
  // snapshotSerializers: [],

  // The test environment that will be used for testing
  // testEnvironment: "jest-environment-jsdom",

  // Options that will be passed to the testEnvironment
  // testEnvironmentOptions: {},

  // Adds a location field to test results
  // testLocationInResults: false,

  // The glob patterns Jest uses to detect test files
  // testMatch: [
  //   "**/__tests__/**/*.[jt]s?(x)",
  //   "**/?(*.)+(spec|test).[tj]s?(x)"
  // ],

  // An array of regexp pattern strings that are matched against all test paths, matched tests are skipped
  // testPathIgnorePatterns: [
  //   "/node_modules/"
  // ],

  // The regexp pattern or array of patterns that Jest uses to detect test files
  // testRegex: [],

  // This option allows the use of a custom results processor
  // testResultsProcessor: null,

  // This option allows use of a custom test runner
  // testRunner: "jasmine2",

  // This option sets the URL for the jsdom environment. It is reflected in properties such as location.href
  // testURL: "http://localhost",

  // Setting this value to "fake" allows the use of fake timers for functions such as "setTimeout"
  // timers: "real",

  // A map from regular expressions to paths to transformers
  "transform": {
    // 用 `vue-jest` 处理 `*.vue` 文件
    ".*\\.(vue)$": "vue-jest",
    "^.+\\.js$": "<rootDir>/node_modules/babel-jest"
  }

  // An array of regexp pattern strings that are matched against all source file paths, matched files will skip transformation
  // transformIgnorePatterns: [
  //   "/node_modules/"
  // ],

  // An array of regexp pattern strings that are matched against all modules before the module loader will automatically return a mock for them
  // unmockedModulePathPatterns: undefined,

  // Indicates whether each individual test should be reported during the run
  // verbose: null,

  // An array of regexp patterns that are matched against all source file paths before re-running tests in watch mode
  // watchPathIgnorePatterns: [],

  // Whether to use watchman for file crawling
  // watchman: true,
};


lib/index.umd.js
(function webpackUniversalModuleDefinition(root, factory) {
	if(typeof exports === 'object' && typeof module === 'object')
		module.exports = factory();
	else if(typeof define === 'function' && define.amd)
		define([], factory);
	else if(typeof exports === 'object')
		exports["index"] = factory();
	else
		root["index"] = factory();
})((typeof self !== 'undefined' ? self : this), function() {
return /******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = "fb15");
/******/ })
/************************************************************************/
/******/ ({

/***/ "f6fd":
/***/ (function(module, exports) {

// document.currentScript polyfill by Adam Miller

// MIT license

(function(document){
  var currentScript = "currentScript",
      scripts = document.getElementsByTagName('script'); // Live NodeList collection

  // If browser needs currentScript polyfill, add get currentScript() to the document object
  if (!(currentScript in document)) {
    Object.defineProperty(document, currentScript, {
      get: function(){

        // IE 6-10 supports script readyState
        // IE 10+ support stack trace
        try { throw new Error(); }
        catch (err) {

          // Find the second match for the "at" string to get file src url from stack.
          // Specifically works with the format of stack traces in IE.
          var i, res = ((/.*at [^\(]*\((.*):.+:.+\)$/ig).exec(err.stack) || [false])[1];

          // For all scripts on the page, if src matches or if ready state is interactive, return the script tag
          for(i in scripts){
            if(scripts[i].src == res || scripts[i].readyState == "interactive"){
              return scripts[i];
            }
          }

          // If no match, return null
          return null;
        }
      }
    });
  }
})(document);


/***/ }),

/***/ "fb15":
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);

// CONCATENATED MODULE: ./node_modules/@vue/cli-service/lib/commands/build/setPublicPath.js
// This file is imported into lib/wc client bundles.

if (typeof window !== 'undefined') {
  if (true) {
    __webpack_require__("f6fd")
  }

  var i
  if ((i = window.document.currentScript) && (i = i.src.match(/(.+\/)[^/]+\.js(\?.*)?$/))) {
    __webpack_require__.p = i[1] // eslint-disable-line
  }
}

// Indicate to webpack that this file can be concatenated
/* harmony default export */ var setPublicPath = (null);

// CONCATENATED MODULE: ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f8e3fa32-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/index.vue?vue&type=template&id=214d1c89&
var render = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return (_vm.render)?_c('div',{staticClass:"nu_dialog_wrap",class:_vm.computedClass},[_vm._t("mask",[_c('label',{staticClass:"nu_dialog_mask",on:{"click":_vm.handleClickMask}})]),_vm._t("dialog",[_c('div',{staticClass:"nu_dialog"},[_vm._t("close",[_c('label',{staticClass:"nu_dialog_close",on:{"click":_vm.handleClickClose}},[_vm._t("close-icon",[_vm._v("×")])],2)]),_vm._t("default")],2)])],2):_vm._e()}
var staticRenderFns = []


// CONCATENATED MODULE: ./src/index.vue?vue&type=template&id=214d1c89&

// CONCATENATED MODULE: ./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/index.vue?vue&type=script&lang=js&
//
//
//
//
//
//
//
//
//
//
//
//
//
//
//
//
//
//

/* harmony default export */ var lib_vue_loader_options_srcvue_type_script_lang_js_ = ({
    name: 'NuDialog',
    data() {
        return {
            show: false,
            render: this.open
        }
    },
    props: {
        isPortal: {
            type: Boolean,
            default: true
        },
        open: Boolean,
        position: {
            type: String,
            default: 'middle'
        },
        beforeClose: Function,
        speed: {
            type: Number,
            default: 200
        }
    },
    watch: {
        open(val) {
            val ? this.handleOpen() : this.handleClose();
        }
    },
    mounted() {
        this.open && this.handleOpen();
    },
    destroyed() {
        // if appendToBody is true, remove DOM node after destroy
        if (this.$el && this.$el.parentNode) {
            this.$el.parentNode.removeChild(this.$el);
        }
    },
    computed: {
        computedClass() {
            const ret = {};

            // 是否显示弹窗
            if (this.show) {
                ret._open = true;
            }

            // 设置弹窗方向
            if (this.position) {
                ret['_' + this.position] = true;
            }
            return ret;
        }
    },
    methods: {
        handleOpen() {
            this.render = true;
            this.isPortal && document.body.appendChild(this.$el);
            // 动画的时间
            setTimeout(() => {
                this.show = true;
            }, 16);
            document.body.classList.add('nu_dialog_open');
        },
        dialogClose() {
            this.show = false;
            // 动画的时间
            setTimeout(() => {
                this.render = false;
            }, this.speed);
            document.body.classList.remove('nu_dialog_open');
        },
        handleClose() {
            if (typeof this.beforeClose === 'function') {
                const ret = this.beforeClose();
                /* 只有当用户的返回值为false 的时候才会阻止弹窗关闭 */
                if (ret !== false) {
                    this.dialogClose();
                }
            } else {
                this.dialogClose();
            }
        },
        /**
         * 当遮罩点击当时候触发
         * @param e
         */
        handleClickMask(e) {
            e.preventDefault();
            this.$emit('onClickMask');
        },
        /**
         * 当关闭按钮点击的时候触发
         * @param e
         */
        handleClickClose(e) {
            e.preventDefault();
            this.$emit('update:open', false);
        }
    }
});

// CONCATENATED MODULE: ./src/index.vue?vue&type=script&lang=js&
 /* harmony default export */ var srcvue_type_script_lang_js_ = (lib_vue_loader_options_srcvue_type_script_lang_js_); 
// CONCATENATED MODULE: ./node_modules/vue-loader/lib/runtime/componentNormalizer.js
/* globals __VUE_SSR_CONTEXT__ */

// IMPORTANT: Do NOT use ES2015 features in this file (except for modules).
// This module is a runtime utility for cleaner component module output and will
// be included in the final webpack user bundle.

function normalizeComponent (
  scriptExports,
  render,
  staticRenderFns,
  functionalTemplate,
  injectStyles,
  scopeId,
  moduleIdentifier, /* server only */
  shadowMode /* vue-cli only */
) {
  // Vue.extend constructor export interop
  var options = typeof scriptExports === 'function'
    ? scriptExports.options
    : scriptExports

  // render functions
  if (render) {
    options.render = render
    options.staticRenderFns = staticRenderFns
    options._compiled = true
  }

  // functional template
  if (functionalTemplate) {
    options.functional = true
  }

  // scopedId
  if (scopeId) {
    options._scopeId = 'data-v-' + scopeId
  }

  var hook
  if (moduleIdentifier) { // server build
    hook = function (context) {
      // 2.3 injection
      context =
        context || // cached call
        (this.$vnode && this.$vnode.ssrContext) || // stateful
        (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext) // functional
      // 2.2 with runInNewContext: true
      if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
        context = __VUE_SSR_CONTEXT__
      }
      // inject component styles
      if (injectStyles) {
        injectStyles.call(this, context)
      }
      // register component module identifier for async chunk inferrence
      if (context && context._registeredComponents) {
        context._registeredComponents.add(moduleIdentifier)
      }
    }
    // used by ssr in case component is cached and beforeCreate
    // never gets called
    options._ssrRegister = hook
  } else if (injectStyles) {
    hook = shadowMode
      ? function () { injectStyles.call(this, this.$root.$options.shadowRoot) }
      : injectStyles
  }

  if (hook) {
    if (options.functional) {
      // for template-only hot-reload because in that case the render fn doesn't
      // go through the normalizer
      options._injectStyles = hook
      // register for functioal component in vue file
      var originalRender = options.render
      options.render = function renderWithStyleInjection (h, context) {
        hook.call(context)
        return originalRender(h, context)
      }
    } else {
      // inject component registration as beforeCreate hook
      var existing = options.beforeCreate
      options.beforeCreate = existing
        ? [].concat(existing, hook)
        : [hook]
    }
  }

  return {
    exports: scriptExports,
    options: options
  }
}

// CONCATENATED MODULE: ./src/index.vue





/* normalize component */

var component = normalizeComponent(
  srcvue_type_script_lang_js_,
  render,
  staticRenderFns,
  false,
  null,
  null,
  null
  
)

/* harmony default export */ var src = (component.exports);
// CONCATENATED MODULE: ./src/index.js


/* istanbul ignore next */
src.install = function(Vue) {
    Vue.component(src.name, src);
};

/* harmony default export */ var src_0 = (src);

// CONCATENATED MODULE: ./node_modules/@vue/cli-service/lib/commands/build/entry-lib.js


/* harmony default export */ var entry_lib = __webpack_exports__["default"] = (src_0);



/***/ })

/******/ });
});
//# sourceMappingURL=index.umd.js.map